### PR TITLE
feat(pack): add pack-gzip-level

### LINKED ISSUE #6393: [`pnpm pack`] Add option to allow setting custom compression level
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

**Describe the user story**

vercel/next.js team uses `pnpm pack` in their tests and our packages contain some rust binaries, which are large and slow to pack. We want to decrease the compression level used in packing to speed this process up. The fastest compression level (1) could pack our packages under `.5s`, but the default level (6) takes up to `9s`.

Some folks may prefer increasing the compression level to achieve a smaller packed size.

**Describe the solution you'd like**

Anything that could expose this option to the user would work.
For example, `tar` uses `GZIP=-1` environment variable, but we could probably add a command argument or anything else.

It could be quickly done with a single line here: https://github.com/pnpm/pnpm/blob/5a5512f14c47f4778b8d2b6d957fb12c7ef40127/releasing/plugin-commands-publishing/src/pack.ts#L165

**Describe the drawbacks of your solution**

This is a specific use case that many folks might not need.

**Describe alternatives you've considered**

None

### PR DESCRIPTION
**Pull request by jankaifer**
This options allows specifying custom compression level.
Underlying node implementation enforces correct number ranges,
but silently ignores `NaN`, so we explicitly check that provided value is valid.

More information about this `level` option can be found in [node docs](https://nodejs.org/api/zlib.html#class-options) and [zlib specs](https://zlib.net/manual.html#Advanced).

Supported values are (`6` is default in node):
```c
// Compression levels.
#define Z_NO_COMPRESSION         0
#define Z_BEST_SPEED             1
#define Z_BEST_COMPRESSION       9
#define Z_DEFAULT_COMPRESSION  (-1)
```

I didn't add any test but I tested manually
that compressing with `-1`, `0` and `1` create an installable package.

Also tested manually that using `pnpm config set pack-gzip-level 0` and `--pack-gzip-level=0` both work as expected.

fix #6393

### CODE REVIEW

File: releasing/plugin-commands-publishing/src/pack.ts [hunk 1]
```diff
@@ -127,6 +127,21 @@ async function readReadmeFile (filesMap: Record<string, string>) {
...
 }
 
+function getGzipConfig () {
+  const gzipConfig: ZlibOptions = {}
+
+  const level = process.env.PNPM_PACK_GZIP_LEVEL
+  if (level !== undefined) {
+    const parsedLevel = parseInt(level)
+    if (isNaN(parsedLevel)) {
+      throw new PnpmError('INVALID_GZIP_LEVEL', `Invalid gzip level: '${level}'`)
```
Comment (jankaifer) [inline releasing/plugin-commands-publishing/src/pack.ts]: Let me know if I need to register this error somewhere.

A quick full-text search didn't yield exactly one instance of other errors.

File: releasing/plugin-commands-publishing/src/pack.ts [hunk 2]
```diff
@@ -127,6 +127,21 @@ async function readReadmeFile (filesMap: Record<string, string>) {
   return readmeFile
 }
 
+function getGzipConfig () {
+  const gzipConfig: ZlibOptions = {}
+
+  const level = process.env.PNPM_PACK_GZIP_LEVEL
```
Comment (zkochan) [inline releasing/plugin-commands-publishing/src/pack.ts]: This should be a new setting in `@pnpm/config` and passed through options.

Comment (jankaifer) [reply releasing/plugin-commands-publishing/src/pack.ts]: Just figured out how `@pnpm/config` works when I saw your commit.

Also thanks for adding the changelog entry, completely forgot about that!

**Review by jankaifer — COMMENTED** [review]
I think you forgot to pass that option here:
https://github.com/pnpm/pnpm/pull/6406/files#diff-c518d24f9667ad9a2c03aaa643610621010dcfb05670c326948311d7088bab10R115

**Review by zkochan — APPROVED** [review]

**Comment by jankaifer** [comment]
Thanks a bunch for your help. I appreciate it.

File: releasing/plugin-commands-publishing/src/pack.ts [hunk 3]
```diff
@@ -30,6 +30,9 @@ export function rcOptionsTypes () {
 export function cliOptionsTypes () {
   return {
     'pack-destination': String,
+    ...pick([
+      'pack-gzip-level',
```
Comment (nipunn1313) [inline releasing/plugin-commands-publishing/src/pack.ts:34]: Consider also adding this to `help()` below so it shows up in `pnpm pack --help` for discoverability?

Comment (nhuray) [reply releasing/plugin-commands-publishing/src/pack.ts:34]: @zkochan I noticed the documentation [mentioned](https://pnpm.io/cli/pack)  the option `--pack-gzip-level` BUT the help does not display it ! 

I also noticed the property `packGzipLevel` is missing in [publish.ts#handler](https://github.com/pnpm/pnpm/blob/main/releasing/plugin-commands-publishing/src/publish.ts#L123) (but present in [publish.ts#publish](https://github.com/pnpm/pnpm/blob/main/releasing/plugin-commands-publishing/src/publish.ts#L144))

**Pull request merged**